autoprotocol/protocol.py

```diff
@@ -1035,102 +1046,253 @@ def liquid_handle_dispense(
-        liquid = _validate_as_instance(liquid, LiquidClass)
-        method._liquid = liquid
-
-        # prime volume
-        if method.prime is True:
-            prime = method.default_prime(None)
-        elif method.prime is False:
-            prime = Unit(0, "uL")
+        # Check destinations match the number of sources
+        len_src = len(source)
+        if not isinstance(destination, list):
+            if isinstance(destination, (WellGroup, Well)):
+                destination: List[WellGroup] = [WellGroup(destination) for _ in source]
+            else:
+                raise ValueError(
+                    f"The destination given {type(destination)} {destination} is not a WellGroup or Well"
+                )
+        else:
+            if len_src > 1:
+                len_dests = len(destination)
+                if len_dests != len_src:
+                    raise ValueError(
+                        f"When specifying more than one destination locations "
+                        f"the number of destinations must match the number of sources "
+                        f"num of destinations {len_dests} != num of sources {len_src}"
+                    )
+                destination: List[WellGroup] = [WellGroup(d) for d in destination]
+
+                for (sw, number_of_chips), destinations in zip(source, destination):
+                    # Check that the number of destinations match the number of chips specified for each
+                    # source destination mapping
+                    if len(destinations) != number_of_chips:
+                        raise ValueError(
+                            f"The number of destinations {len(destinations)} does not match the number of "
+                            f"chips {number_of_chips} specified for the given source {(sw, number_of_chips)}"
+                        )
+
+        len_dests = len(destination)
+        if not isinstance(volume, list):
+            volume: List[Unit] = [
+                [parse_unit(v, "uL") for v in [volume] * len(wg)] for wg in destination
+            ]
         else:
-            prime = method.prime
-        prime_volume = parse_unit(prime, "uL")
-
-        # predispense volume
-        if method.predispense is True:
-            predispense = method.default_predispense(None)
-        elif method.predispense is False:
-            predispense = Unit(0, "uL")
+            len_vols = len(volume)
+            if len_vols != len_dests:
+                raise ValueError(
+                    f"When specifying more than one volume "
+                    f"the number of volumes must match the number of destination locations "
+                    f"num of volumes {len_vols} != num of destinations {len_dests}"
+                )
+            volume: List[Unit] = [
+                [parse_unit(v, "uL") for v in [vol] * len(destination[i])]
+                for i, vol in enumerate(volume)
+            ]
```

Comment:
the way this is written currently will raise an error if the volumes are passed in as a list of 1 while there are more than one destination. 

not sure it makes a huge difference but passing in a list of 1 could be treated the same as a single volume.